File: 3046_Split_the_Array/solution.c
bool isPossibleToSplit(int* nums, int numsSize) {
    int freq[101] = {0};
    for (int i = 0; i < numsSize; i++) freq[nums[i]]++;
    for (int i = 0; i < 101; i++) if (freq[i] > 2) return false;
    return true;
}

File: 3046_Split_the_Array/solution.py
class Solution:
    def isPossibleToSplit(self, nums: List[int]) -> bool:
        freq = {}
        for num in nums:
            freq[num] = freq.get(num, 0) + 1

        for value in freq.values():
            if value > 2:
                return False

        return True
